src/cpu/cpu.rs

```diff
@@ -116,7 +116,7 @@ impl Cpu {
                         let rs = self.read_reg_gpr(instr.rs()) as u32;
                         let rt = self.read_reg_gpr(instr.rt()) as u32;
 
-                        let res = ((rs * rt) as i32) as u64;
+                        let res = (rs as u64) * (rt as u64);
 
                         // TODO: Undefined if last 2 instructions were
```

Comment:
Yeah, re the undefined behavior it's correct enough as-is (or rather will be after accepting this PR) but I think I want to keep the TODO around as a mental reminder to possibly print out a warning as @DavidEGrayson suggests. Thanks for the extra info as well :)